### PR TITLE
Replaced deprecated PackageInfo.signatures with PackageInfo.signingInfo

### PR DESCRIPTION
**Pull request by paulkagiri**
Replaced the deprecated `PackageInfo.signatures` with `PackageInfo.signingInfo`. Created a utility method across code and tests for compatibility reasons.

Work Item: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1226178
Issue: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/851

### CODE REVIEW

**Review by gitaumoses4 — APPROVED** [review]

File: package-inspector/src/main/java/com/microsoft/inspector/MainActivity.java [hunk 1]
```diff
@@ -203,4 +204,29 @@ private String getAuthenticatorAppMetadata(@NonNull final String pkgName) {
     private boolean isAnAuthenticatorApp(@NonNull final String pkgName) {
         return mPkgAuthenticators.containsKey(pkgName);
     }
+
+    private Signature[] getSignatures(PackageInfo packageInfo) {
```
Comment (iambmelt) [inline package-inspector/src/main/java/com/microsoft/inspector/MainActivity.java:208]:
```suggestion
    private Signature[] getSignatures(@Nullable final PackageInfo packageInfo) {
```

Please mark variables `final` where possible, and annotate expectations around null/non-null

File: package-inspector/src/main/java/com/microsoft/inspector/MainActivity.java [hunk 2]
```diff
@@ -126,15 +127,15 @@ public void onItemClick(AdapterView<?> adapterView, View view, int position, lon
...
+                            getPackageManagerFlag()
                     );
 
                     String packageSigningSha = "";
 
-                    if (null != packageInfo
-                            && null != packageInfo.signatures
-                            && packageInfo.signatures.length > 0) {
-                        final Signature signature = packageInfo.signatures[0];
+                    Signature [] signatures = getSignatures(packageInfo);
```
Comment (iambmelt) [inline package-inspector/src/main/java/com/microsoft/inspector/MainActivity.java:135]:
```suggestion
                    final Signature [] signatures = getSignatures(packageInfo);
```

File: package-inspector/src/main/java/com/microsoft/inspector/MainActivity.java [hunk 3]
```diff
@@ -203,4 +204,29 @@ private String getAuthenticatorAppMetadata(@NonNull final String pkgName) {
...
+                return packageInfo.signingInfo.getApkContentsSigners();
+            } else {
+                return packageInfo.signingInfo.getSigningCertificateHistory();
+            }
+        }
+
+        return packageInfo.signatures;
+    }
+
+    private int getPackageManagerFlag() {
```
Comment (iambmelt) [inline package-inspector/src/main/java/com/microsoft/inspector/MainActivity.java:225]: Method can be `static`?

**Comment by iambmelt** [comment]
There seems to be an instrumented test failing -- have you reviewed it?

**Review by iambmelt — APPROVED** [review]

**Pull request merged**